staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness_test.go

```diff
@@ -324,49 +327,30 @@ func TestApfCancelWaitRequest(t *testing.T) {
 		t.Error(err)
 	}
 
-	checkForExpectedMetricsWithRetry(t, []string{
+	checkForExpectedMetrics(t, []string{
 		"apiserver_current_inflight_requests",
 		"apiserver_request_terminations_total",
 		"apiserver_dropped_requests_total",
 	})
 }
 
-// wait async metrics to be collected
-func checkForExpectedMetricsWithRetry(t *testing.T, expectedMetrics []string) {
-	maxRetries := 5
-	var checkErrors []error
-	for i := 0; i < maxRetries; i++ {
-		t.Logf("Check for expected metrics with retry %d", i)
-		metricsFamily, err := legacyregistry.DefaultGatherer.Gather()
-		if err != nil {
-			t.Fatalf("Failed to gather metrics %v", err)
-		}
-
-		metrics := map[string]interface{}{}
-		for _, mf := range metricsFamily {
-			metrics[*mf.Name] = mf
-		}
-
-		checkErrors = checkForExpectedMetrics(expectedMetrics, metrics)
-		if checkErrors == nil {
-			return
-		}
-
-		time.Sleep(1 * time.Second)
+// gathers and checks the metrics.
+func checkForExpectedMetrics(t *testing.T, expectedMetrics []string) {
+	metricsFamily, err := legacyregistry.DefaultGatherer.Gather()
+	if err != nil {
+		t.Fatalf("Failed to gather metrics %v", err)
 	}
-	for _, checkError := range checkErrors {
-		t.Error(checkError)
+
+	metrics := map[string]interface{}{}
+	for _, mf := range metricsFamily {
+		metrics[*mf.Name] = mf
 	}
-}
 
-func checkForExpectedMetrics(expectedMetrics []string, metrics map[string]interface{}) []error {
-	var errs []error
 	for _, metricName := range expectedMetrics {
 		if _, ok := metrics[metricName]; !ok {
 			if !ok {
-				errs = append(errs, errors.New("Scraped metrics did not include expected metric "+metricName))
+				t.Errorf(errors.New("Scraped metrics did not include expected metric " + metricName))
```

Comment:
```
$ go test -race .
# k8s.io/apiserver/pkg/server/filters [k8s.io/apiserver/pkg/server/filters.test]
./priority-and-fairness_test.go:352:24: cannot use "errors".New("Scraped metrics did not include expected metric " + metricName) (type error) as type string in argument to t.common.Errorf
```

The first argument of `Errorf` is a format string.

Reply:
yes, I was suggesting something like `t.Errorf("scraped metrics did not include expected metric %q", metricName)`